staging/src/k8s.io/client-go/util/flowcontrol/backoff.go

```diff
@@ -133,22 +145,28 @@ func (p *Backoff) IsInBackOffSinceUpdate(id string, eventTime time.Time) bool {
 	if !ok {
 		return false
 	}
-	if hasExpired(eventTime, entry.lastUpdate, p.maxDuration) {
+	if p.hasExpired(eventTime, entry.lastUpdate, p.maxDuration) {
 		return false
 	}
 	return eventTime.Sub(entry.lastUpdate) < entry.backoff
 }
 
-// Garbage collect records that have aged past maxDuration. Backoff users are expected
-// to invoke this periodically.
+// Garbage collect records that have aged past GCTimeout or, if nil,
+// 2*maxDuration. Backoff users are expected to invoke this periodically.
 func (p *Backoff) GC() {
 	p.Lock()
 	defer p.Unlock()
 	now := p.Clock.Now()
 	for id, entry := range p.perItemBackoff {
-		if now.Sub(entry.lastUpdate) > p.maxDuration*2 {
-			// GC when entry has not been updated for 2*maxDuration
-			delete(p.perItemBackoff, id)
+		if p.GCTimeout > 0 {
```

Comment:
Do we need the GC timeout here, or can we just reuse the `hasExpired` function? The old logic was equivalent to calling `hasExpired(now, entry.lastUpdate, p.maxDuration)`. If there's a custom `HasExpiredFunc`, is there any reason not to use the same logic?

Reply:
I see your point, where `eventTime` = `now` for hasExpired. I will make this change which will tidy up the amount of configuratino. Thanks!